### ISSUE TITLE
Accessibility: not able to navigate to "view cart" via keyboard

### ISSUE THREAD

**Issue by helgatheviking** [woocommerce/woocommerce#21025]
After a product is added to the cart, a user is unable to navigate to the "view cart" link in the added to cart message via keyboard. This makes it difficult for visually impaired users to go directly to the cart after adding a product.

Here's the full text provided to me as part of a name your price ticket, but I think this pertains to core.

----
I have 2 blind singers who navigate our site using keyboard only (mostly using tab, no mouse) and vision-impaired software. They can successfully navigate to the Name Your Price edit box and change the amount, and then tab to the ADD TO CART button and hit return. This then pops up a box at the top of the page that the item has been added to the cart with a button VIEW CART. They cannot navigate to the VIEW CART button.

If you try this using the TAB key only, you see it skips over the VIEW CART button. I have found that they can navigate to the CART menu item in the left menu and get to the cart that way. But this is not something a vision-impaired user could navigate without intervention; in other words they'd get stuck and have to ask for help.

Short Term: can you make it so the tab key goes to the View Cart pop up.

Long Term: Can you improve wooCommerce functioning for vision-impaired users. Having a pop-up box is very difficult to navigate if you can't see it happening.

**Comment by rodrigoprimo** [comment]
> I believe she was referring to the single product page add to cart message:

Thanks for the clarification, @helgatheviking. I spoke with @claudiulodro, and he will check this issue as he knows more accessibility than I do.

**Comment by claudiulodro** [comment]
I've confirmed that tabbing to the button currently works, but it takes a lot of tabbing to reach it. The best solution would be to add the attribute `tabindex="1"` to the "View cart" button in the notice. Then it will be the first thing that you tab to when the notice is present.

The tricky part of implementing this straightforward change is that notices are run through `wp_kses_post`, and it doesn't allow `tabindex` attributes.

**Comment by rodrigoprimo** [comment]
Related WP core tickets:

https://core.trac.wordpress.org/ticket/39288
https://core.trac.wordpress.org/ticket/30421

Maybe we can use `wp_kses()` instead of `wp_kses_posts()`  and add `tabindex` to the list of allowed attributes?

**Comment by peterfabian** [comment]
Since there is a reload when adding to cart from single product page, I'm not entirely sure that adding tabindex=1 is the best solution, but from one point of view, the notices are there to be noticed and interacted with first if needed, so here we go: #21108.
